packages/query-core/src/queryCache.ts

```diff
@@ -166,8 +166,21 @@ export class QueryCache extends Subscribable<QueryCacheListener> {
   }
 
   find<TQueryFnData = unknown, TError = unknown, TData = TQueryFnData>(
-    arg1: QueryKey,
-    arg2?: QueryFilters,
+    filters: QueryFilters,
+  ): Query<TQueryFnData, TError, TData> | undefined
+  /**
+   * @deprecated This method should be used with only one object argument.
+   */
+  find<TQueryFnData = unknown, TError = unknown, TData = TQueryFnData>(
+    queryKey: QueryKey,
+    filters?: OmitKeyof<QueryFilters, 'queryKey'>,
+  ): Query<TQueryFnData, TError, TData> | undefined
+  /**
+   * @deprecated This method should be used with only one object argument.
+   */
+  find<TQueryFnData = unknown, TError = unknown, TData = TQueryFnData>(
+    arg1: QueryKey | QueryFilters,
+    arg2?: OmitKeyof<QueryFilters, 'queryKey'>,
```

Comment:
I tagged jsdoc @deprecated on unnecessary type overriding of QueryCache's methods -- find, findAll